core/src/main/java/org/apache/calcite/sql/type/OperandTypes.java

```diff
@@ -1225,6 +1229,64 @@ private static class MapFromEntriesOperandTypeChecker
     }
   }
 
+  /**
+   * Operand type-checking strategy for a ARRAY function, it allows empty array.
+   *
+   * <p> The reason it overrides SameOperandTypeChecker#checkOperandTypesImpl is that it needs
+   * to handle the scenario where row/struct type and NULL exist simultaneously in array.
+   * This scenario need be supported, but will be rejected by the current checkOperandTypesImpl.
```

Comment:
I think the last row of this comment should be removed. This would be appropriate in the PR, but not in a comment.

Reply:
yes, removed.